### PR TITLE
UPDATE 1.4.0

### PR DESCRIPTION
**Pull request by NotRealPaz**
```diff
+ fix music modules
+ update all dependencies to lastest
+ update readme
- remove ffmpeg-binaries
```

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -21,17 +20,20 @@ let prefix = config.commandPrefix;
 const chatlog = client.channels.get(config.chatlogidroom);
 
 // Import Music controller
-const musiccontroller = require('./music/musiccontroller.js');
-const music = new musiccontroller(client);
+const musiccontrollersrc = require('./musicmodules/musiccontroller');
```
Comment (icharge) [inline index.js]: musicControllerSrc

File: index.js [hunk 2]
```diff
@@ -21,17 +20,20 @@ let prefix = config.commandPrefix;
...
+const musicontroller = new musiccontrollersrc(client, prefix);
```
Comment (icharge) [inline index.js]: musicController

File: index.js [hunk 3]
```diff
@@ -56,36 +58,65 @@ client.once('ready', () => {
...
-    .setFooter("**ถ้าไม่ทำงานแสดงว่าคุณลงผิดที่ ลองลงใหม่")
+    .setFooter("**ถ้าไม่ทำงานแสดงว่าคุณลงผิดที่ ลองลงใหม่");
 });
 
-client.on('message', async message => {
+client.on('message', async (message) => {
   if (message.author.bot) return;
-  if (message.channel.type != "text") return;
+  if (message.channel.type !== "text") return;
   let messageArray = message.content.split(/\s+/g);
```
Comment (icharge) [inline index.js]: Use `const messageArray` instead

File: index.js [hunk 4]
```diff
@@ -56,36 +58,65 @@ client.once('ready', () => {
...
-      message.channel.send({
-        embed: yurinetEmbed
-      });
-    };
-    if (cmd === "game") {
-      message.channel.send({
-        embed: gameEmbed
-      });
-    };
+  if (!messageArray[0].startsWith(prefix)) return;
```
Comment (icharge) [inline index.js]: You should check `messageArray` is valid first before accessing array index

File: index.js [hunk 5]
```diff
@@ -56,36 +58,65 @@ client.once('ready', () => {
...
+  let cmd = messageArray[0].slice(prefix.length).toLowerCase();
+  let cmd = command.slice(prefix.length);
```
Comment (icharge) [inline index.js]: Extract this getting command string into **function**.

File: index.js [hunk 6]
```diff
@@ -56,36 +58,65 @@ client.once('ready', () => {
...
+  if (cmd === "ddraw") {
+    message.channel.send({
+      embed: DDwrapperEmbed
+    });
+    return;
+  };
+  if (cmd === "yurinet") {
```
Comment (icharge) [inline index.js]: use `if ... else` in same variable.
**if** statements are have no **;** (semi-colon)

File: index.js [hunk 7]
```diff
@@ -95,60 +126,41 @@ const io = require('socket.io-client');
...
+socket.on('lobby.chat', async (response) => {
+  let ChatDate = datetime(response.timestamp);
 
   function getchatmsg() {
     return '``' + response.name + '``' + " : " + response.message;
   };
 
   function formatChatMessage() {
-    return '``' + getChatDate() + "`` " + getchatmsg();
+    return '``' + ChatDate + "`` " + getchatmsg();
```
Comment (icharge) [inline index.js]: use back tick `  for concatenation strings

``return `'${ChatDate}' ${getchatmsg()}`;``

**Review by icharge — COMMENTED** [review]
Read comments

**Comment by NotRealPaz** [comment]
I change everything you requested except this [comments](https://github.com/icharge/yurinet-discord-bot/pull/8#discussion_r207956049).

**Comment by NotRealPaz** [comment]
u check my code everytim

File: index.js [hunk 8]
```diff
@@ -5,88 +5,102 @@
...
+    musicController.showqueue(message);
+  } else if (cmd === "stop") {
+    message.delete();
+    musicController.checkafkbot(message);
+    musicController.stop(message);
+  } else if (cmd === "skip") {
+    message.delete();
+    musicController.checkafkbot(message);
+    musicController.skip(message);
   };
```
Comment (icharge) [inline index.js]: If else block has no semicolon (;)

File: modules/musiccontroller.js [hunk 9]
```diff
@@ -0,0 +1,213 @@
...
+const search = require("./search.js");
+const checker = require("./checker.js");
+const EventEmitter = require('eventemitter3');
+class musiccontroller extends EventEmitter {
+    constructor() {
+        super();
+        this.connection = {};
+        this.autoLeaveTimeout = [];
+        this.isplaying = {};
+    };
```
Comment (icharge) [inline modules/musiccontroller.js]: No semi colon

File: modules/musiccontroller.js [hunk 10]
```diff
@@ -0,0 +1,213 @@
...
+    async addsong(msg, song) {
+        this.cleartime(msg);
+        let guildid = msg.guild.id;
+        if (!this.isplaying[guildid]) {
+            this.isplaying[guildid] = false;
+        };
```
Comment (icharge) [inline modules/musiccontroller.js]: Please correct all of IF..ELSE block.
NO SEMICOLONS

File: modules/musiccontroller.js [hunk 11]
```diff
@@ -0,0 +1,213 @@
...
+                this.addsong(msg);
+            }, 1000);
+            return;
+        };
+    };
+
+    async cleartime(msg) {
+        let guildid = msg.guild.id;
+        clearTimeout(this.autoLeaveTimeout[guildid]);
+    };
```
Comment (icharge) [inline modules/musiccontroller.js]: Methods have no semicolons

**Comment by NotRealPaz** [comment]
removed 33 semicolon

**Review by icharge — APPROVED** [review]
Go ahead

**Pull request merged**